src/event_scanner/modes/historic.rs

```diff
@@ -0,0 +1,208 @@
+        modes::common::{ConsumerMode, handle_stream},
+    },
+};
+
+pub struct HistoricScannerBuilder {
+    block_range_scanner: BlockRangeScanner,
+    // Defaults to Earliest
+    from_block: BlockNumberOrTag,
+    // Defaults to Latest
+    to_block: BlockNumberOrTag,
+}
+
+pub struct HistoricEventScanner<N: Network> {
+    config: HistoricScannerBuilder,
+    block_range_scanner: ConnectedBlockRangeScanner<N>,
+    listeners: Vec<EventListener>,
+}
+
+impl HistoricScannerBuilder {
+    pub(super) fn new() -> Self {
+        Self {
+            block_range_scanner: BlockRangeScanner::new(),
+            from_block: BlockNumberOrTag::Earliest,
+            to_block: BlockNumberOrTag::Latest,
+        }
+    }
+
+    #[must_use]
+    pub fn max_block_range(mut self, max_block_range: u64) -> Self {
+        self.block_range_scanner.max_block_range = max_block_range;
+        self
+    }
+
+    #[must_use]
+    pub fn from_block(mut self, block: impl Into<BlockNumberOrTag>) -> Self {
+        self.from_block = block.into();
+        self
+    }
+
+    #[must_use]
+    pub fn to_block(mut self, block: impl Into<BlockNumberOrTag>) -> Self {
+        self.to_block = block.into();
+        self
+    }
+
+    /// Connects to the provider via WebSocket.
+    ///
+    /// Final builder method: consumes the builder and returns the actual `HistoricEventScanner`.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the connection fails
+    pub async fn connect_ws<N: Network>(
+        self,
+        ws_url: Url,
+    ) -> TransportResult<HistoricEventScanner<N>> {
+        let block_range_scanner = self.block_range_scanner.connect_ws::<N>(ws_url).await?;
+        Ok(HistoricEventScanner { config: self, block_range_scanner, listeners: Vec::new() })
+    }
+
+    /// Connects to the provider via IPC.
+    ///
+    /// Final builder method: consumes the builder and returns the actual `HistoricEventScanner`.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the connection fails
+    pub async fn connect_ipc<N: Network>(
+        self,
+        ipc_path: String,
+    ) -> TransportResult<HistoricEventScanner<N>> {
+        let block_range_scanner = self.block_range_scanner.connect_ipc::<N>(ipc_path).await?;
+        Ok(HistoricEventScanner { config: self, block_range_scanner, listeners: Vec::new() })
+    }
+
+    /// Connects to an existing provider.
+    ///
+    /// Final builder method: consumes the builder and returns the actual `HistoricEventScanner`.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the connection fails
+    #[must_use]
+    pub fn connect<N: Network>(self, provider: RootProvider<N>) -> HistoricEventScanner<N> {
+        let block_range_scanner = self.block_range_scanner.connect::<N>(provider);
+        HistoricEventScanner { config: self, block_range_scanner, listeners: Vec::new() }
+    }
+}
+
+impl<N: Network> HistoricEventScanner<N> {
+    pub fn create_event_stream(
```

Comment:
```suggestion
    pub fn subscribe(
```
this might be clear enough, wdyt?